crates/subspace-networking/src/node_runner.rs

```diff
@@ -137,6 +141,17 @@ impl NodeRunner {
             }
             SwarmEvent::NewListenAddr { address, .. } => {
                 self.shared.listeners.lock().push(address.clone());
+                if matches!(address.iter().next(), Some(Protocol::Memory(_))) {
+                    // This is necessary for local connections using circuit relay
+                    if self.is_relay_server {
+                        self.swarm
+                            .add_external_address(address.clone(), AddressScore::Infinite);
+                    }
+                } else {
+                    // TODO: Add support for public address for add_external_address, AutoNAT
+                    self.swarm
```

Comment:
Why do we need this if we won't use non-memory relay transport?

Reply:
There is still a need for node to know if it is reachable in some ways from the outside. For memory transport we only care about relay, but for others, we need to specify something like `/ip4/12.9.54.79/tcp/1234/p2p/Qm1//p2p-circuit/p2p/Qm2` so that other peers on the network know how to reach this node or else everyone will be able to make outgoing connections, but no one will be able to know where to connect to. This is the address shared with other nodes on the network on libp2p level

Reply:
Did your tests fail without that? IIRC relay servers routed requests successfully on my initial tests: `relayed-requests` doesn't contain tcp address as an external however is able to successfully get a result. Am I missing something?

Reply:
Removed in last commit